### ISSUE TITLE
ngZoneEventCoalescing causes page refresh while navigating to different route

### ISSUE THREAD

**Issue by HimanshuGoel** [angular/angular#48831]
### Which @angular/* package(s) are the source of the bug?

core

### Is this a regression?

No

### Description

While passing the `ngZoneEventCoalescing` set as `true` in the `BootstrapOptions` as per the [documentation](https://angular.io/api/core/BootstrapOptions), it causes the page refresh issue while navigating to the next route.

Suppose the current URL is - `http://localhost:4200/parent`

If I click on a button to navigate to another route i.e. `http://localhost:4200/test-page`, it first goes to `http://localhost:4200/test-page?` by appending a question mark (?) at the end of the URL, which triggers a complete page reload and finally gets navigated to `http://localhost:4200/test-page` route.

If I pass `ngZoneEventCoalescing` as `false` then it works fine.

### Please provide a link to a minimal reproduction of the bug

https://stackblitz.com/github/HimanshuGoel/issue-reproduction3?file=src%2Fmain.ts

When we click the "Primary" button, we navigate to the next route i.e. "test-page" and the application performs the complete reload.

However, in the `main.ts` file, if we set the `ngZoneEventCoalescing` flag as `false`, the application navigates to the "test-page" and doesn't reload the application which is the expected behaviour.

In summary - Setting the `ngZoneEventCoalescing` flag to `true` appears to break the router navigation.

### Please provide the exception or error you saw

```true
There is no error occurring in the console.
```

### Please provide the environment you discovered this bug in (run `ng version`)

```true
Angular CLI - 14.2.2
Angular Core - 14.2.0

This issue also occurs with Angular 15 but StackBlitz has an issue with the Angular 15 applications running on their platform.

Please refer to this repo branch for more information - https://github.com/HimanshuGoel/issue-reproduction3
```

**Comment by JeanMeche** [comment]
You're not having a problem with `ngZoneEventCoalescing ` but with a form submit. 

`type="submit" (click)="onClick()"`

When clicking the button your are triggering the native HTML submit.

**Comment by HimanshuGoel** [comment]
Thanks, @JeanMeche for your response.

I believe it is an issue with the `ngZoneEventCoalescing` flag, I have amended the issue post to more accurately reflect the problem. Also, provided the revised reproduction URL as it seems to be some issue with StackBlitz while running an Angular 15 application.

The button is no longer a submit button and simply changing the `ngZoneEventCoalescing` flag from `true` to `false` fixes the issue of the application reloading when routing to the 'test-page'.

https://stackblitz.com/github/HimanshuGoel/issue-reproduction3?file=src%2Fmain.ts

**Comment by JeanMeche** [comment]
By default; a button in a form is of `type="submit"`, so you'll have to add `type="button"` you won't have that problem. 

You're still showing a form submit problem in your 2nd repro.

**Comment by HimanshuGoel** [comment]
I didn't feel any need to explicitly use of  "type='button'" attribute within a form in an Angular application for a such routing scenario.

Additionally, I am still seeking clarification as to why the implementation of the `ngZoneEventCoalescing` flag affects the behaviour of routing.

**Comment by JeanMeche** [comment]
It has to do with the changeDetection running later when you coaslesce. 
You can see that chrome logs a warning when you don't coalesce and the change detection is triggered directly on the navigation: 
`Form submission canceled because the form is not connected`. 

You'll see that you also have the reloading without the coalescing when you use a timeout : 

```
  onClick() {
    setTimeout(() => {
      this.router.navigate(['test-page']);
    })
  }
```

When you coalesce, all the code has the time to run before the change detectiong (which will remove the form). So the navigator can still trigger de redirection on submit.

**Comment by SopraTests** [comment]
@JeanMeche  OK, that all makes sense, thanks.
We have a dynamically rendered Angular application (based on JSON configuration) so are quite sensitive to lots of change detection cycles. We would like to enable the ngZoneEventCoalescing flag to help reduce this. However, we also would like to have a button of type submit on a screen so that a user can hit the Enter key to submit the form data.

So, what would be the recommended way of resolving this?

**Comment by JeanMeche** [comment]
Use `type="button"` when you don't want buttons to submit the forms  👍

**Comment by SopraTests** [comment]
But we do want a submit button so that a user can use the Enter key to submit the form without having to tab to that button.
We also want ngZoneEventCoalescing. These 2 things don't seem to play nicely together?

**Comment by JeanMeche** [comment]
For that you'll need to implement `(submit)="onSubmit()"` on the form and don't forget to return `false` to prevent the `POST`!

**Comment by pkozlowski-opensource** [comment]
Hello, we reviewed this issue and determined that it doesn't fall into the bug report or feature request category. This issue tracker is not suitable for support requests, please repost your issue on [StackOverflow](http://stackoverflow.com/) using tag `angular`.

If you are wondering why we don't resolve support issues via the issue tracker, please [check out this explanation](https://github.com/angular/angular/blob/master/CONTRIBUTING.md#-got-a-question-or-problem).

**Comment by pkozlowski-opensource** [comment]
Thnx for helping out @JeanMeche
